modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIEndpointRelevantObjectEntryModelListener.java

```diff
@@ -282,13 +287,43 @@ private void _validateSingleElementPath(
 		}
 	}
 
+	private boolean _validateUniquePathParameter(
+			long companyId, String pathParameter, long responseAPISchemaId)
+		throws Exception {
+
+		if (Objects.equals(pathParameter, "id") ||
+			Objects.equals(pathParameter, "externalReferenceCode")) {
+
+			return true;
+		}
+
+		ObjectEntry responseAPISchemaObjectEntry =
+			_objectEntryLocalService.getObjectEntry(responseAPISchemaId);
+
+		Map<String, Serializable> values =
+			responseAPISchemaObjectEntry.getValues();
+
+		List<String> uniqueObjectFields =
+			_objectEntryHelper.getUniqueObjectFieldNames(
+				companyId, (String)values.get("mainObjectDefinitionERC"));
+
+		if (uniqueObjectFields.contains(pathParameter)) {
```

Comment:
I am not sure if this is rigth, as we are comparing an internal field name with an external field name (I guess it is external the one that is configured in the path)